lci_strategic_plugin/src/lci_strategic_plugin_algo.cpp

```diff
@@ -0,0 +1,1291 @@
+  if (v1 > v0)
+    distances.dx2 = ((pow(v1, 2) - pow(v0, 2)) / (2 * a_max));
+  else
+    distances.dx2 = ((pow(v1, 2) - pow(v0, 2)) / (2 * a_min));
+
+  distances.dx3 = ((pow(v_min, 2) - pow(v0, 2)) / (2 * a_min)) + ((pow(v1, 2) - pow(v_min, 2)) / (2 * a_max));
+  distances.dx4 = ((pow(v_min, 2) - pow(v0, 2)) / (2 * a_min));
+  distances.dx5 = - pow(v0, 2) / (2 * a_min);
+
+  return distances;
+}
+
+void LCIStrategicPlugin::print_params(TrajectoryParams params)
+{
+  ROS_DEBUG_STREAM("\n");
+  ROS_DEBUG_STREAM("t0: " << std::to_string(params.t0_));
+  ROS_DEBUG_STREAM("v0: " << params.v0_);
+  ROS_DEBUG_STREAM("x0: " << params.x0_);
+
+  ROS_DEBUG_STREAM("t1: " << std::to_string(params.t1_));
+  ROS_DEBUG_STREAM("v1: " << params.v1_);
+  ROS_DEBUG_STREAM("x1: " << params.x1_);
+  ROS_DEBUG_STREAM("a1: " << params.a1_);
+
+  ROS_DEBUG_STREAM("t2: " << std::to_string(params.t2_));
+  ROS_DEBUG_STREAM("v2: " << params.v2_);
+  ROS_DEBUG_STREAM("x2: " << params.x2_);
+  ROS_DEBUG_STREAM("a2: " << params.a2_);
+
+  ROS_DEBUG_STREAM("t3: " << std::to_string(params.t3_));
+  ROS_DEBUG_STREAM("v3: " << params.v3_);
+  ROS_DEBUG_STREAM("x3: " << params.x3_);
+  ROS_DEBUG_STREAM("a3: " << params.a3_);
+
+  ROS_DEBUG_STREAM("\n");
+}
+
+void LCIStrategicPlugin::print_boundary_distances(BoundaryDistances delta_xs)
+{
+  ROS_DEBUG_STREAM("\n");
+  ROS_DEBUG_STREAM("dx1: " << delta_xs.dx1);
+  ROS_DEBUG_STREAM("dx2: " << delta_xs.dx2);
+  ROS_DEBUG_STREAM("dx3: " << delta_xs.dx3);
+  ROS_DEBUG_STREAM("dx4: " << delta_xs.dx4);
+  ROS_DEBUG_STREAM("dx5: " << delta_xs.dx5 << "\n");
+}
+
+std::vector<TrajectoryParams> LCIStrategicPlugin::get_boundary_traj_params(double t, double v0, double v1, double v_max, double v_min, double a_max, double a_min, double x0, double x_end, double dx, BoundaryDistances boundary_distances)
+{
+  double dx1 = boundary_distances.dx1;
+  double dx2 = boundary_distances.dx2;
+  double dx3 = boundary_distances.dx3;
+  double dx4 = boundary_distances.dx4;
+  double dx5 = boundary_distances.dx5;
+  TrajectoryParams traj1, traj2, traj3, traj4, traj5, traj6, traj7, traj8;
+
+  // t1, t2, t3
+  if (dx < dx2)
+  {
+    traj1 = boundary_accel_or_decel_incomplete_upper(t, v0, v1, a_max, a_min, x0, x_end, dx);
+    traj2 = traj1;
+    traj3 = traj1;
+  }
+  else if (dx < dx1)
+  {
+    traj1 = boundary_accel_nocruise_notmaxspeed_decel(t, v0, v1, a_max, a_min, x0, x_end, dx);
+    traj2 = traj1;
+    traj3 = boundary_accel_or_decel_complete_upper(t, v0, v1, x0, x_end, dx);
+  }
+  else
+  {
+    traj1 = boundary_accel_cruise_maxspeed_decel(t, v0, v1, v_max, a_max, a_min, x0, x_end, dx);
+    traj2 = boundary_accel_nocruise_maxspeed_decel(t, v0, v1, v_max, a_max, a_min, x0, x_end, dx);
+    traj3 = boundary_accel_or_decel_complete_upper(t, v0, v1, x0, x_end, dx);
+  }
+  // t4, t5, t6, t7
+  if (dx < dx4)
+  {
+    traj4 = boundary_decel_incomplete_lower(t, v0, a_min, x0, x_end, dx);
+    traj5 = traj4;
+    traj6 = traj4;
+    traj7 = traj4;
```

Comment:
Please change these to:

traj4 = traj1;
traj5 = traj1;
traj6 = boundary_decel_incomplete_lower(t, v0, a_min, x0, x_end, dx);
traj7 = traj6;